ynput/OpenPype
# Publisher: Explicit save

## Changelog Description
Publisher have explicit button to save changes, so reset can happen without saving any changes. Save still happens automatically when publishing is started or on publisher window close. But a popup is shown if context of host has changed. Important context was enhanced by workfile path (if host integration supports it) so workfile changes are captured too. In that case a dialog with confirmation is shown to user. All callbacks that may require save of context were moved to main window to be able handle dialog show at one place. Save changes now returns success so the rest of logic is skipped -> publishing won't start, when save of instances fails.

Save and reset buttons have shortcuts (Ctrl + s and Ctrls + r).

## Additional info
Messages shown in dialogs may be cryptic for artists.

## Testing notes:
1. Open host using Publisher
2. Open Publisher tool
3. Create an instance
4. Change any attribute
5. Hit refresh
6. Attribute values should be changed back to their origin values
7. Change any attribute
8. Hit save
9. Hit refresh
10. Attribute values should stay
11. Change any attribute (e.g. `review`)
12. Hit publish
13. Value of the attribute should be propagated to publishing (review is created)
